datafusion/functions-nested/src/flatten.rs

```diff
@@ -77,7 +77,6 @@ impl Flatten {
     pub fn new() -> Self {
         Self {
             signature: Signature {
-                // TODO (https://github.com/apache/datafusion/issues/13757) flatten should be single-step, not recursive
                 type_signature: TypeSignature::ArraySignature(
                     ArrayFunctionSignature::RecursiveArray,
```

Comment:
Is this signature still applicable?

Maybe we should switch to Array, and mark RecursiveArray as deprecated?